optimum/amd/ryzenai/modeling_decoder.py

```diff
@@ -0,0 +1,412 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+# Licensed under the MIT License.
+"""RyzenAIModelForCausalLM classes, allowing to run ONNX Models with ONNX Runtime VITIS-AI EP using the same API as Transformers."""
+
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import onnxruntime as ort
+import torch
+
+from optimum.utils import NormalizedConfigManager, check_if_transformers_greater
+from transformers import (
+    AutoModelForCausalLM,
+    GenerationConfig,
+)
+from transformers.modeling_outputs import CausalLMOutputWithPast
+
+from .modeling import RyzenAIModel
+from .utils import set_builtins, set_environment_variables
+
+
+if TYPE_CHECKING:
+    from transformers import PretrainedConfig
+
+if check_if_transformers_greater("4.25.0"):
```

Comment:
We can probably require `transformers>=4.26`, this is >1y old

Reply:
Done